invenio_github/assets/semantic-ui/js/invenio_github/index.js

```diff
@@ -0,0 +1,85 @@
+// This file is part of InvenioGithub
+// Copyright (C) 2023 CERN.
+//
+// Invenio Github is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+const sync_button = document.getElementById("sync_repos");
+if (sync_button) {
+  sync_button.addEventListener("click", function () {
+    const resultMessage = document.getElementById("sync-result-message");
+    const loaderIcon = document.getElementById("loaderIcon");
+    loaderIcon.classList.add("loading");
+
+    const url = "/api/user/github/repositories/sync";
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    fetch(url, options).then((response) => {
+      loaderIcon.classList.remove("loading");
+      if (response.ok) {
+        resultMessage.classList.remove("hidden");
+        resultMessage.classList.add("basic");
+        resultMessage.classList.add("green");
+        resultMessage.textContent = "Repositories synced successfully";
+        setTimeout(function () {
+          resultMessage.classList.add("hidden");
+        }, 5000);
+      } else {
+        resultMessage.classList.remove("hidden");
+        resultMessage.classList.add("basic");
+        resultMessage.classList.add("red");
+        resultMessage.textContent = `Request failed with status code: ${response.status}`;
+        setTimeout(function () {
+          resultMessage.classList.add("hidden");
+        }, 5000);
+      }
+    });
+  });
+}
+
+const repositories = document.getElementsByClassName("repositories-list");
+if (repositories) {
+  for (const repo of repositories) {
+    repo.addEventListener("change", function (event) {
+      sendEnableRequest(event.target.checked, repo);
+    });
+  }
+}
+
+function sendEnableRequest(checked, repo) {
+  const repo_id = repo.querySelector("input[data-repo-id]").getAttribute("data-repo-id");
+  const switchMessage = repo.querySelector("#repo-switch-message");
+
+  let url;
+  if (checked === true) {
+    url = "/api/user/github/repositories/" + repo_id + "/enable";
+  } else {
+    if (checked === false) {
+      url = "/api/user/github/repositories/" + repo_id + "/disable";
+    }
+  }
+
+  const options = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  };
+
+  fetch(url, options).then((response) => {
```

Comment:
I think we need to improve the error handling because we don't catch the cases that there is a network error, see https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful . Also for readability maybe is better to use `async...await` syntax...